File: source/src/main/java/org/rss/tools/mpl/domain/Document.java
package org.rss.tools.mpl.domain;

import java.util.ArrayList;
import java.util.List;

/**
 * Represents root document
 */
public class Document {

	private Document tempĺate;
	
	private List<String> stateList;
	
	private final List<String> styleList = new ArrayList<>();
	
	private final List<Section> sectionList = new ArrayList<>();
	
	public Document getTempĺate() {
		return tempĺate;
	}

	public void setTempĺate(Document tempĺate) {
		this.tempĺate = tempĺate;
	}

	public List<String> getStateList() {
		return stateList;
	}

	public void setStateList(List<String> stateList) {
		this.stateList = stateList;
	}

	public List<String> getStyleList() {
		return styleList;
	}
	
	public void addStyleFile(String item) {
		styleList.add(item);
	}
	
	public List<Section> getSectionList() {
		return sectionList;
	}

	public void addSection(Section sec) {
		this.sectionList.add(sec);
	}

	@Override
	public String toString() {
		return "Document [tempĺate=" + tempĺate + ", states=" + stateList + ", styles=" + styleList + "]";
	}
	
}

File: source/src/main/kotlin/org/rss/tools/mpl/domain/Document.kt
package org.rss.tools.mpl.domain

import java.util.ArrayList

data class Document(val sectionList: MutableList<Section> = ArrayList()) {

    var template: Document? = null
    var templateRef: String? = null

    val stateList = mutableListOf<String>()

    val styleList = mutableListOf<String>()

    fun addSection(sec: Section) = this.sectionList.add(sec)

    fun addStyleFile(item: String) = styleList.add(item)

    override fun toString(): String {
        return "Document [template=$template, states=$stateList, styles=$styleList]"
    }

}